storage/txcache/txListForSender.go

```diff
@@ -0,0 +1,191 @@
+package txcache
+
+import (
+	linkedList "container/list"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+)
+
+// TxListForSender is
+type TxListForSender struct {
+	Items          *linkedList.List
+	mutex          sync.Mutex
+	copyBatchIndex *linkedList.Element
+	copyBatchSize  int
+	orderNumber    int64
+	sender         string
+}
+
+// TxListForSenderNode is a node of the linked list
+type TxListForSenderNode struct {
+	TxHash []byte
+	Tx     *transaction.Transaction
+}
+
+// NewTxListForSender creates a new (sorted) list of transactions
+func NewTxListForSender(sender string, globalIndex int64) *TxListForSender {
+	return &TxListForSender{
+		Items:       linkedList.New(),
+		orderNumber: globalIndex,
+		sender:      sender,
+	}
+}
+
+// AddTx adds a transaction in sender's list
+// This is a "sorted" insert
+func (list *TxListForSender) AddTx(txHash []byte, tx *transaction.Transaction) {
+	// We don't allow concurent interceptor goroutines to mutate a given sender's list
+	list.mutex.Lock()
+
+	nonce := tx.Nonce
+	mark := list.findTxWithLargerNonce(nonce)
+	newNode := TxListForSenderNode{txHash, tx}
+
+	if mark == nil {
+		list.Items.PushBack(newNode)
+	} else {
+		list.Items.InsertBefore(newNode, mark)
+	}
+
+	list.mutex.Unlock()
+}
+
+func (list *TxListForSender) findTxWithLargerNonce(nonce uint64) *linkedList.Element {
+	for element := list.Items.Front(); element != nil; element = element.Next() {
+		value := element.Value.(TxListForSenderNode)
+		if value.Tx.Nonce > nonce {
+			return element
+		}
+	}
+
+	return nil
+}
+
+// RemoveTx removes a transaction from the sender's list
+func (list *TxListForSender) RemoveTx(tx *transaction.Transaction) {
```

Comment:
use transactionHandler interface instead of defined struct. Add check.ifNil for input parameters for public methods.